pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -188,3 +351,55 @@ func getInfraMachineTemplateFromProvider(platform configv1.PlatformType) (client
 		return nil, fmt.Errorf("%w: %s", errPlatformNotSupported, platform)
 	}
 }
+
+// updateMAPIMachineSetSpecFromCAPI updates the MAPI MachineSet spec based on the CAPI MachineSet.
+func updateMAPIMachineSetSpecFromCAPI(ctx context.Context, mapiMachineSet *machinev1beta1.MachineSet, capiMachineSet *capiv1beta1.MachineSet, newProviderSpec machinev1beta1.ProviderSpec) {
+	logger := log.FromContext(ctx).WithValues("function", "updateMAPIMachineSetSpecFromCAPI")
+
+	// Update the ProviderSpec
+	mapiMachineSet.Spec.Template.Spec.ProviderSpec = newProviderSpec
+
+	// Update labels (CAPI labels take priority)
+	mapiMachineSet.Spec.Template.Labels = util.MergeMaps(mapiMachineSet.Spec.Template.Labels, capiMachineSet.Spec.Template.Labels)
+
+	// Update taints ?? I can't find 'em on the CAPI MachineSet
+
+	// Update Replicas
+	mapiMachineSet.Spec.Replicas = capiMachineSet.Spec.Replicas
+
+	// Update MinReadySeconds
+	mapiMachineSet.Spec.MinReadySeconds = capiMachineSet.Spec.MinReadySeconds
+
+	// Update DeletePolicy
+	mapiMachineSet.Spec.DeletePolicy = capiMachineSet.Spec.DeletePolicy
+
+	// Update Selector
+	mapiMachineSet.Spec.Selector = capiMachineSet.Spec.Selector
+
+	logger.Info("Updated MAPI MachineSet spec from CAPI MachineSet")
+}
+
+// updateMAPIMachineSetStatusFromCAPI updates the MAPI MachineSet status based on the CAPI MachineSet.
+func updateMAPIMachineSetStatusFromCAPI(ctx context.Context, mapiMachineSet *machinev1beta1.MachineSet, capiMachineSet *capiv1beta1.MachineSet) {
```

Comment:
We haven't yet looked at converting status, we were focusing on spec first. When we do get to it, I think this will want to be in the conversion library itself.

Perhaps look at how this might integrate directly into the library?

Reply:
Have discussed, will stash this until we get onto updating the conversion library.